src/main/webapp/feed.js

```diff
@@ -485,7 +488,8 @@ function checkCheckbox(challenge) {
 }
 
 async function updateUserCurrentChallenge(id) {
-    const putRequest = new Request(`/user?chal=${id}`, {method: 'PUT'});
+    let id_token = gapi.auth2.getAuthInstance().currentUser.get().getAuthResponse().id_token;
+    const putRequest = new Request(`/user?id_token=${id_token}&chal=${id}`, {method: 'PUT'});
```

Comment:
nit: Perhaps refactor all your `let id_token = gapi.auth2....` and `const putRequest....` lines to a function makePutRequest(id_token, challenge_id, status) helper method?